token-minter/sources/modules/mint_stage.move

```diff
@@ -37,27 +35,29 @@ module minter::mint_stage {
     /// The amount of tokens should be greater than zero.
     const EINSUFFICIENT_AMOUNT: u64 = 11;
 
-
     #[resource_group_member(group = aptos_framework::object::ObjectGroup)]
     struct MintStageData has key {
-        /// stage ("private", "presale", "public", etc.) to mint stage mapping.
-        mint_stages: SimpleMap<String, MintStage>,
-        /// Stages sorted from the earliest start time.
-        stages: vector<String>,
+        mint_stages: vector<MintStage>,
     }
 
+    /// stage ("private", "presale", "public", etc.) to mint stage mapping.
     struct MintStage has store {
```

Comment:
Have explored those 2 ways and those were my initial design considerations. 
1. Option
This won't work as the `SmartTable` doesn't have the drop ability, and if you initially set the option to `none` and then try to create the value, it will require destructuring the whole struct.
2. Objects
The way you're describing means a new object for a mint stage would be created everytime, which is not needed and a struct is more appropriate for storage and efficiency imo.

Reply:
What does done mean here? The mint stage phases have ended, the mint has ended etc. Also this would need to be a separate transaction deleting all stale stages.

Reply:
Discussed offline, have refactored to use `Object<MintStage>` for a more extendable design.